Makefile

```diff
@@ -77,12 +83,23 @@ freeze-deps:
 	pip-compile -o requirements/dev.txt requirements/dev.in
 	pip-compile -o requirements/docs.txt requirements/docs.in
 
+	cp requirements/app.txt requirements/${PYV}/app.txt
```

Comment:
So you are creating the .txt files we used to have, then copy them over to their version directory.

Why not create them there right away? Shouldn't we stop having top-level .txt files now?

Reply:
Will make the necessary changes. It makes sense to create them directly.

So the current change with this PR might break the CI. I think another PR followed by this is recommended to fix the CI (which builds the Python package using pinned deps)

Reply:
Maybe another commit within this PR would suffice to get the CI going again, even.